WebCrawler/MainWindow.xaml.cs

```diff
@@ -15,14 +28,86 @@
 
 namespace WebCrawler
 {
-    /// <summary>
-    /// Interaction logic for MainWindow.xaml
-    /// </summary>
     public partial class MainWindow : Window
```

Comment:
This code should not be in mainwindow class. Models have no knowledge of view. It needs to be separate as a model and not in the codebehind.